spec/services/spell_filtering_spec.rb

```diff
@@ -0,0 +1,89 @@
+require "rails_helper"
+
+RSpec.describe "spell index can be filtered" do
+  before(:all) do
+    51.times do
+      FactoryGirl.create(:spell)
+    end
+  end
+
+  let!(:level_1_paladin_spell) do
+    FactoryGirl.create(:spell, name: "Level 1 Paladin Spell", class_requirements:
```

Comment:
this class requirements hash is going to get unweildy quick. We should look at how we can pull this out into something less painful

Reply:
In `spec_helper.rb` I put a default hash inside of a `config.before(:all) do` block. So the hash is generated at the start of the test and is called with @class_requirements_template.

Reply:
I was thinking that maybe it should be its own class or something to that effect Don't worry about it for now I'm sure we'll figure some way of handling it

Reply:
I suppose that makes sense, in that we could instantly pass certain class levels to it. The only downside I see to putting it in spec_helper is the memory use for each test (though it's a small hash).